tests/www/views/test_views.py

```diff
@@ -62,6 +62,22 @@ def test_configuration_expose_config(admin_client):
     check_content_in_response(["Airflow Configuration"], resp)
 
 
+@mock.patch.dict(os.environ, {"AIRFLOW__WEBSERVER__CONFIG_FILE": "/tmp/my_custom_webserver_config.py"})
+def test_webserver_configuration_config_file(admin_client):
+    conf = initialize_config()
+    conf.validate()
+
+    assert WEBSERVER_CONFIG == os.path.join(AIRFLOW_HOME, "webserver_config.py")
```

Comment:
I might be wrong but as per my understanding and running the debugger: Initially, the default path is taken. Once we reach lines: https://github.com/apache/airflow/pull/29926/files#diff-2ab05df659c739ff335a407c4a6cddc5f177e9c402aca296991f53fa341f4dfeR1537-R1542, it generates the file to the new path instead.

Reply:
It's good that the assert down below passes (meaning Airflow does create it in the right place). However, we should also test that `WEBSERVER_CONFIG` gets set to the file in the config. You might want to import it here after running initialize?

This also has the potential of side effects. You should switch back to using `mock.patch.dict`, and you should also try and clean up the file as well with a try/finally.

Reply:
> You might want to import it here after running initialize

Not sure how I can do that. Can i get some help here?

Reply:
I think the issue is that the `local_conf.get("webserver", "config_file")` is unable to return the right value of the webserver config file here. This is because the `mock.patch.dict` doesn't seem to add these values correctly for the unit test. Could i get some help here @jedcunningham ?

Reply:
Adding a log statement to print the env variable set and it seems to reflect it there.
```
PASSED           [100%][2023-05-10T14:10:06.620+0530] {test_views.py:68} WARNING - /tmp/my_custom_webserver_config.py
```

Reply:
Hmm, maybe try with `conf_vars` like this instead:
https://github.com/apache/airflow/blob/d6051fd10a0949264098af23ce74c76129cfbcf4/tests/operators/test_email.py#L59

Reply:
Not helpful. Wondering if we can get the main feature in here and then I can create a follow up PR to investigate this in more detail and fix it. @jedcunningham thoughts?

Reply:
I pushed a commit into your branch that fixes the test. Took me a bit of time to get it actually working, but that exercises the config option and doesn't have any side effects (like leaving WEBSERVER_CONFIG with the custom config from this test).